### PR TITLE
Move HttpClientSlim to Microsoft.AspNetCore.Testing (#279)

### PR DESCRIPTION
**Pull request by cesarbs**
#279

### CODE REVIEW

File: src/Microsoft.AspNetCore.Testing/HttpClientSlim.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+using System.Net.Http;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    // Lightweight version of HttpClient implemented using Socket and SslStream
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNetCore.Testing/HttpClientSlim.cs]: If you're going to have this comment why not make it a doc comment?

File: src/Microsoft.AspNetCore.Testing/Microsoft.AspNetCore.Testing.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
   <PropertyGroup>
     <Description>Various helpers for writing tests that use ASP.NET Core.</Description>
-    <TargetFrameworks>netstandard2.0;net46;net461</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
```
Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Testing/Microsoft.AspNetCore.Testing.csproj]: Remind me again why we can remove net46 now but we couldn't a while back? https://github.com/aspnet/Testing/commit/089c3b87c6f31a074271d08624e7bf875f8db696.

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Testing/Microsoft.AspNetCore.Testing.csproj]: This is what I asked on Teams. I don't know. But I can't move the tests here without removing that. If you think I shouldn't remove `net46`, I'll `#if` some stuff out before merging.

Comment (JunTaoLuo) [inline src/Microsoft.AspNetCore.Testing/Microsoft.AspNetCore.Testing.csproj]: I would hold off on the change on removing net46. As pointed out by @natemcmaster, it seems like Razor still cross-compiles net46 in its packages and tests (https://github.com/aspnet/Razor/search?utf8=%E2%9C%93&q=net46&type=). Removing net46 will probably break it. Let's go with the extra `#if`s for now.

**Comment by cesarbs** [comment]
@JunTaoLuo Actually I can bring the tests over, given I created a new test project for them. It wasn't possible (without ifs) with the exists test projects.

Will be sending reaction PRs soon.

**Pull request merged**